src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/PublishIntegrationTest.cs

```diff
@@ -60,6 +60,53 @@ public async Task Publish_PublishesAssembly()
             Assert.FileCountEquals(result, 0, Path.Combine(PublishOutputPath, "Views"), "*.cshtml");
         }
 
+        [Fact]
+        [InitializeTestProject("SimpleMvcWithContent")]
+        public async Task Publish_PublishesContent()
+        {
+            var result = await DotnetMSBuild("Publish");
+
+            Assert.BuildPassed(result);
+
+            Assert.FileExists(result, OutputPath, "SimpleMvcWithContent.dll");
+            Assert.FileExists(result, OutputPath, "SimpleMvcWithContent.pdb");
+            Assert.FileExists(result, OutputPath, "SimpleMvcWithContent.Views.dll");
+            Assert.FileExists(result, OutputPath, "SimpleMvcWithContent.Views.pdb");
+            Assert.FileExists(result, OutputPath, "appsettings.json");
+            Assert.FileExists(result, OutputPath, "appsettings.development.json");
+            Assert.FileDoesNotExist(result, OutputPath, Path.Combine("Properties", "launchSettings.json"));
+
+            Assert.FileExists(result, PublishOutputPath, "SimpleMvcWithContent.dll");
+            Assert.FileExists(result, PublishOutputPath, "SimpleMvcWithContent.pdb");
+            Assert.FileExists(result, PublishOutputPath, "SimpleMvcWithContent.Views.dll");
+            Assert.FileExists(result, PublishOutputPath, "SimpleMvcWithContent.Views.pdb");
+            Assert.FileExists(result, PublishOutputPath, "appsettings.json");
+            Assert.FileExists(result, PublishOutputPath, "appsettings.development.json");
+            Assert.FileDoesNotExist(result, PublishOutputPath, Path.Combine("Properties", "launchSettings.json"));
+
+            // By default refs and .cshtml files will not be copied on publish
+            Assert.FileCountEquals(result, 0, Path.Combine(PublishOutputPath, "refs"), "*.dll");
+            Assert.FileCountEquals(result, 0, Path.Combine(PublishOutputPath, "Views"), "*.cshtml");
+        }
+
+        [Fact(Skip = "Single file publish is currently broken")]
```

Comment:
@rynowak any ideas about how we can test these types of scenarios? I get this:

```
[7/12/2019 10:46:58 PM Warning] Test run will use DLL(s) built for framework .NETCoreApp,Version=v1.0 and platform X86. Following DLL(s) do not match framework/platform settings.
Microsoft.NET.Sdk.Razor.Test.dll is built for Framework 3.0 and Platform AnyCPU.
Go to http://go.microsoft.com/fwlink/?LinkID=236877&clcid=0x409 for more details on managing these settings.

[7/12/2019 10:46:58 PM Informational] [xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.1 (64-bit .NET Core 3.0.0-preview6-27804-01)
[7/12/2019 10:46:59 PM Informational] [xUnit.net 00:00:00.73]   Discovering: Microsoft.NET.Sdk.Razor.Test
[7/12/2019 10:46:59 PM Informational] [xUnit.net 00:00:00.80]   Discovered:  Microsoft.NET.Sdk.Razor.Test
[7/12/2019 10:46:59 PM Informational] [xUnit.net 00:00:00.81]   Starting:    Microsoft.NET.Sdk.Razor.Test
[7/12/2019 10:47:02 PM Error] [xUnit.net 00:00:03.07]     PublishSingleFile_PublishesContentOutsideOfSingleFile [FAIL]
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       Microsoft.AspNetCore.Razor.Design.IntegrationTests.Assert+BuildFailedException : Build failed.
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       C:\dev\git\AspNetCore-Tooling\.dotnet\dotnet.exe msbuild /nr:false /bl /p:RunningAsTest=true /p:MicrosoftNETCoreApp30PackageVersion=3.0.0-preview8-27912-09 /p:RuntimeAdditionalRestoreSources=C:\dev\git\AspNetCore-Tooling\src\Razor\test\Microsoft.NET.Sdk.Razor.Test\..\testapps\TestPackageRestoreSource /p:_RazorBuildServerPipeName="31899d46-1f1c-496d-95aa-4d832110076e" /t:Restore /t:Publish /p:Configuration=Debug /p:PublishSingleFile=true /p:RuntimeIdentifier=win-x64
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       Microsoft (R) Build Engine version 16.2.0-preview-19278-01+d635043bd for .NET Core
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       Copyright (C) Microsoft Corporation. All rights reserved.
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       C:\dev\git\AspNetCore-Tooling\.dotnet\sdk\3.0.100-preview6-012264\MSBuild.dll -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\dev\git\AspNetCore-Tooling\.dotnet\sdk\3.0.100-preview6-012264\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\dev\git\AspNetCore-Tooling\.dotnet\sdk\3.0.100-preview6-012264\dotnet.dll -maxcpucount -verbosity:m /bl /nr:false /p:RunningAsTest=true /p:MicrosoftNETCoreApp30PackageVersion=3.0.0-preview8-27912-09 /p:RuntimeAdditionalRestoreSources=C:\dev\git\AspNetCore-Tooling\src\Razor\test\Microsoft.NET.Sdk.Razor.Test\..\testapps\TestPackageRestoreSource /p:_RazorBuildServerPipeName=31899d46-1f1c-496d-95aa-4d832110076e /p:Configuration=Debug /p:PublishSingleFile=true /p:RuntimeIdentifier=win-x64 /t:Restore /t:Publish .\SimpleMvcWithContent.csproj
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       C:\Users\david\AppData\Local\Temp\Razor\3zrcuqyu.j5w\SimpleMvcWithContent\SimpleMvcWithContent.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.App.Runtime.win-x64 with version (= 3.0.0-preview8-27912-09)
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       C:\Users\david\AppData\Local\Temp\Razor\3zrcuqyu.j5w\SimpleMvcWithContent\SimpleMvcWithContent.csproj : error NU1102:   - Found 24 version(s) in dotnet-core [ Nearest version: 3.0.0-preview4-19121-14 ]
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       C:\Users\david\AppData\Local\Temp\Razor\3zrcuqyu.j5w\SimpleMvcWithContent\SimpleMvcWithContent.csproj : error NU1102:   - Found 3 version(s) in nuget.org [ Nearest version: 3.0.0-preview6.19307.2 ]
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]       C:\Users\david\AppData\Local\Temp\Razor\3zrcuqyu.j5w\SimpleMvcWithContent\SimpleMvcWithContent.csproj : error NU1102:   - Found 0 version(s) in aspnet-extensions
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.07]         Restore failed in 472.22 ms for C:\Users\david\AppData\Local\Temp\Razor\3zrcuqyu.j5w\SimpleMvcWithContent\SimpleMvcWithContent.csproj.
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.08]       
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.08]       
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.08]       
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.08]       Exit Code:1
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.08]       Stack Trace:
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.08]         C:\dev\git\AspNetCore-Tooling\src\Razor\test\Microsoft.NET.Sdk.Razor.Test\IntegrationTests\Assert.cs(41,0): at Microsoft.AspNetCore.Razor.Design.IntegrationTests.Assert.BuildPassed(MSBuildResult result, Boolean allowWarnings)
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.08]         C:\dev\git\AspNetCore-Tooling\src\Razor\test\Microsoft.NET.Sdk.Razor.Test\IntegrationTests\PublishIntegrationTest.cs(98,0): at Microsoft.AspNetCore.Razor.Design.IntegrationTests.PublishIntegrationTest.PublishSingleFile_PublishesContentOutsideOfSingleFile()
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.08]         --- End of stack trace from previous location where exception was thrown ---
[7/12/2019 10:47:02 PM Informational] [xUnit.net 00:00:03.24]   Finished:    Microsoft.NET.Sdk.Razor.Test
[7/12/2019 10:47:02 PM Informational] ========== Run test finished: 1 run (0:00:04.1880653) ==========
```

Reply:
Did some digging and I’m not sure the current infra is setup for this. Seems like there are issues around how it determines the version of the runtime

Reply:
@pranavkm - would know more about this. Is the problem that the project wasn't restored with the version of the runtime used by the SDK under test?

Reply:
AH, I think the problem is that we don't reference `Microsoft.AspNetCore.App`.

This will never really work. The problem is that the primary thing we do here is iterate on the compiler - that requires us to separate the runtime used by these projects from the **real** runtime that's in the SDK we're using the repo. Basically we need to preserve the ability to make breaking changes to the compiler, that make it incompatible with any version of `M.A.App` that exists.

Reply:
What @rynowak said. You might be able to test this by pinning this to an already released runtime that's available on NuGet.org. That said, a scenario like this is likely much more suited for the CTI. You really want the cross-matrix of the latest runtime + latest SDK which these tests are a poor substitute for.

Reply:
I'll scrap the test and make sure CTI validate.

Reply:
cc @mkArtakMSFT